compiler/optimizer/abstractinterpreter/InliningMethodSummary.cpp

```diff
@@ -104,6 +104,12 @@ const char* TR::PotentialOptimizationPredicate::getName()
 
 bool TR::PotentialOptimizationVPPredicate::holdPartialOrderRelation(TR::VPConstraint* valueConstraint, TR::VPConstraint* testConstraint)  
    {
+   TR_ASSERT_FATAL(testConstraint->isClassObject() != TR_yes, "testConstraint is not a class Object");
+   TR_ASSERT_FATAL(testConstraint->getClass() != NULL, "testConstraint class is Null");
+   TR_ASSERT_FATAL(testConstraint->isNonNullObject(), "testConstraint is a Null");
+   TR_ASSERT_FATAL(testConstraint->getPreexistence() == NULL, "testConstraint already exists");
+   TR_ASSERT_FATAL(testConstraint->getArrayInfo() == NULL, "testConstraint Array Info is NUll");
+   TR_ASSERT_FATAL(testConstraint->getObjectLocation() == NULL, "testContraint object location is Null");
```

Comment:
Some of these assertion failure messages are phrased so that they describe the passing case, but they should describe the failing case instead, like so:
- `testConstraint` is ~not~ a class object
- `testConstraint` has unexpected array info
- `testConstraint` has an unexpected location

Additionally, some of the others are unclear. I would suggest:
- `testConstraint` unexpectedly admits null
- `testConstraint` has unexpected pre-existence info

It's possible though that the `isNonNullObject()` assertion won't remain if the constraint for `checkcast` stops [unnecessarily requiring non-null](https://github.com/eclipse/omr/pull/5508/#discussion_r660076874)

Reply:
> It's possible though that the `isNonNullObject()` assertion won't remain if the constraint for `checkcast` stops [unnecessarily requiring non-null](https://github.com/eclipse/omr/pull/5508/#discussion_r660076874)

Ah, I see the case for a bare `VPClassType` (sufficient for `checkcast`) is already there. And in fact, we might have an integer constraint, which definitely won't satisfy `isNonNullObject()` (and that makes me suspect that these assertions haven't run very much...)

These assertions belong in the case that handles `testConstraint->asClassType()` (i.e. `instanceof`), since that's the code that seems to be making these assumptions about the `testConstraint`, and that's the code I was commenting on when I suggested the assertions. Please move them